nilearn/plotting/tests/test_surf_plotting.py

```diff
@@ -197,9 +197,141 @@ def test_plot_surf_roi_error():
+    plot_img_on_surf(nii, hemispheres=['right', 'left'], views='lateral',
+                     surf_mesh='fsaverage')
+    plot_img_on_surf(nii, hemispheres=['right', 'left'], views='lateral',
+                     surf_mesh='fsaverage5')
+    surf_mesh = fetch_surf_fsaverage()
+    plot_img_on_surf(nii, hemispheres=['right', 'left'], views='lateral',
+                     surf_mesh=surf_mesh)
+
+
+def test_plot_img_on_surf_with_invalid_orientation():
+    kwargs = {"hemisphere": "right", "inflate": True}
+    nii = _generate_img()
+    with pytest.raises(ValueError):
+        plot_img_on_surf(nii, views='latral', **kwargs)
+    with pytest.raises(ValueError):
+        plot_img_on_surf(nii, views=['dorsal', 'post'], **kwargs)
+    with pytest.raises(ValueError):
+        plot_img_on_surf(nii, views=0, **kwargs)
+    with pytest.raises(ValueError):
+        plot_img_on_surf(nii, views=['medial', {'a': 'a'}], **kwargs)
+
+
+def test_plot_img_on_surf_with_invalid_hemisphere():
+    nii = _generate_img()
+    with pytest.raises(ValueError):
+        plot_img_on_surf(
+            nii, views='lateral', inflate=True, hemispheres="lft"
+        )
+    with pytest.raises(ValueError):
+        plot_img_on_surf(
+            nii, views='medial', inflate=True, hemispheres=['lef']
+        )
+    with pytest.raises(ValueError):
+        plot_img_on_surf(
+            nii,
+            views=['anterior', 'posterior'],
+            inflate=True,
+            hemispheres=['left', 'right', 'middle']
+        )
+
+
+def test_plot_img_on_surf_with_figure_kwarg():
+    nii = _generate_img()
+    with pytest.raises(ValueError):
+        plot_img_on_surf(
+            nii,
+            views="anterior",
+            hemispheres="right",
+            figure=True,
+        )
+
+
+def test_plot_img_on_surf_with_axes_kwarg():
+    nii = _generate_img()
+    with pytest.raises(ValueError):
+        plot_img_on_surf(
+            nii,
+            views="anterior",
+            hemispheres="right",
+            inflat=True,
+            axes="something",
+        )
+
+
+def test_plot_img_on_surf_title():
+    nii = _generate_img()
+    title = "Title"
+    fig, axes = plot_img_on_surf(
+        nii, hemispheres='right', views='lateral'
+    )
+    assert fig._suptitle is None, "Created title without title kwarg."
+    fig, axes = plot_img_on_surf(
+        nii, hemispheres='right', views='lateral', title=title
+    )
+    assert fig._suptitle is not None, "Title not created."
+    assert fig._suptitle.get_text() == title, "Title text not assigned."
+
+
+def test_plot_img_on_surf_output_file():
+    nii = _generate_img()
+    fname = Path('check.png').absolute()
+    return_value = plot_img_on_surf(
+        nii, hemispheres='right', views='lateral', output_file=str(fname)
+    )
+    assert return_value is None, "Returned figure and axes on file output."
+    assert fname.is_file(), "Saved image file could not be found."
+    fname.unlink()
```

Comment:
if the test fails the file won't be removed

Reply:
I added a try-except block to remove the file even if the tests fail.